fe/fe-core/src/main/java/com/starrocks/rpc/BackendServiceProxy.java

```diff
@@ -67,9 +67,14 @@ private BackendServiceProxy() {
         if (Config.enable_brpc_share_channel) {
             rpcOptions.setShareThreadPoolUnderEachProxy(true);
             rpcOptions.setShareChannelPool(true);
-            rpcOptions.setThreadPoolSize(1);
+            rpcOptions.setMaxTotoal(Config.brpc_number_of_concurrent_requests_processed_for_share_channel);
+            // After the RPC request sending, the connection will be closed,
+            // if the number of total connections is greater than MaxIdleSize.
+            // Therefore, MaxIdleSize shouldn't less than MaxTotal.
+            rpcOptions.setMaxIdleSize(Config.brpc_number_of_concurrent_requests_processed_for_share_channel);
         } else {
-            rpcOptions.setThreadPoolSize(Config.brpc_number_of_concurrent_requests_processed);
+            rpcOptions.setMaxTotoal(Config.brpc_number_of_concurrent_requests_processed);
+            rpcOptions.setMaxIdleSize(Config.brpc_number_of_concurrent_requests_processed);
```

Comment:
Is there any way to reduce the number of RPCs through Batch or other methods?

Reply:
Yes, maybe we could a some single-flight logic in the future.